Invalid suggest for methods in auto trait

Given the following [code](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3d78bc0ac4aa5d4ce69d876c285f6112)


```rust
#![feature(auto_traits)]

auto trait Foo {
    fn g(&self);
}

trait Bar  {
    fn f(&self) {
        self.g();
    }
}

fn main() {}

```

The current output is:

```
error[E0380]: auto traits cannot have associated items
 --> src/main.rs:4:8
  |
3 | auto trait Foo {
  |            --- auto trait cannot have associated items
4 |     fn g(&self);
  |     ---^-------- help: remove these associated items

error[E0554]: `#![feature]` may not be used on the stable release channel
 --> src/main.rs:1:1
  |
1 | #![feature(auto_traits)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^

error[E0599]: the method `g` exists for reference `&Self`, but its trait bounds were not satisfied
 --> src/main.rs:9:14
  |
9 |         self.g();
  |              ^
  |
note: trait bound `&Self: Foo` was not satisfied
  = note: the following trait bounds were not satisfied:
          `Self: Foo`
          which is required by `&Self: Foo`
  = help: items from traits can only be used if the type parameter is bounded by the trait
help: the following trait defines an item `g`, perhaps you need to add a supertrait for it:
  |
7 | trait Bar: Foo  {
  |          +++++

Some errors have detailed explanations: E0380, E0554, E0599.
For more information about an error, try `rustc --explain E0380`.
error: could not compile `playground` due to 3 previous errors
```

The second error is not reasonable, since we don't allow associated items in auto trait.
Ideally the output should look like:

```
error[E0599]: no method named `g` found for reference `&Self` in the current scope
 --> ./p/e.rs:9:14
  |
9 |         self.g();
  |
```




<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_ASSIGN_START -->

<!-- TRIAGEBOT_ASSIGN_DATA_START$${"user":"chenyukang"}$$TRIAGEBOT_ASSIGN_DATA_END -->

<!-- TRIAGEBOT_ASSIGN_END -->
<!-- TRIAGEBOT_END -->

--- a/compiler/rustc_hir_typeck/src/method/probe.rs
+++ b/compiler/rustc_hir_typeck/src/method/probe.rs
@@ -943,6 +943,9 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
             });
         } else {
             debug_assert!(self.tcx.is_trait(trait_def_id));
+            if self.tcx.trait_is_auto(trait_def_id) {
+                return;
+            }
             for item in self.impl_or_trait_item(trait_def_id) {
                 // Check whether `trait_def_id` defines a method with suitable name.
                 if !self.has_applicable_self(&item) {

--- a/compiler/rustc_hir_typeck/src/method/suggest.rs
+++ b/compiler/rustc_hir_typeck/src/method/suggest.rs
@@ -2306,6 +2306,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         _ => false,
                     }
                 }) && (type_is_local || info.def_id.is_local())
+                    && !self.tcx.trait_is_auto(info.def_id)
                     && self
                         .associated_value(info.def_id, item_name)
                         .filter(|item| {

--- a/src/test/ui/methods/issues/issue-105732.rs
+++ b/src/test/ui/methods/issues/issue-105732.rs
@@ -6,7 +6,8 @@ auto trait Foo {
 
 trait Bar {
     fn f(&self) {
-        self.g(); //~ ERROR the method `g` exists for reference `&Self`, but its trait bounds were not satisfied
+        // issue #105788
+        self.g(); //~ ERROR no method named `g` found for reference `&Self` in the current scope
     }
 }
 

--- a/src/test/ui/methods/issues/issue-105732.stderr
+++ b/src/test/ui/methods/issues/issue-105732.stderr
@@ -6,21 +6,11 @@ LL | auto trait Foo {
 LL |     fn g(&self);
    |     ---^-------- help: remove these associated items
 
-error[E0599]: the method `g` exists for reference `&Self`, but its trait bounds were not satisfied
-  --> $DIR/issue-105732.rs:9:14
+error[E0599]: no method named `g` found for reference `&Self` in the current scope
+  --> $DIR/issue-105732.rs:10:14
    |
 LL |         self.g();
-   |              ^
-   |
-   = note: the following trait bounds were not satisfied:
-           `Self: Foo`
-           which is required by `&Self: Foo`
-           `&Self: Foo`
-   = help: items from traits can only be used if the type parameter is bounded by the trait
-help: the following trait defines an item `g`, perhaps you need to add a supertrait for it:
-   |
-LL | trait Bar: Foo {
-   |          +++++
+   |              ^ help: there is a method with a similar name: `f`
 
 error: aborting due to 2 previous errors
 

